plugins/outputs/wavefront/wavefront.go

```diff
@@ -97,9 +100,16 @@ func (w *Wavefront) Connect() error {
 		connectionURL = senderURLFromHostAndPort(w.Host, w.Port)
 	}
 
+	tlsConfig, err := w.TLSConfig()
+	if err != nil {
+		return err
+	}
+
 	sender, err := wavefront.NewSender(connectionURL,
 		wavefront.BatchSize(w.HTTPMaximumBatchSize),
 		wavefront.FlushIntervalSeconds(flushSeconds),
+		wavefront.TLSConfigOptions(tlsConfig),
+		wavefront.Timeout(time.Duration(w.Timeout)),
```

Comment:
What happens if TLS is disabled? In this case `tlsConfig` will be `nil`... Does that work with `wavefront.TLSConfigOptions` to _not_ use TLS?

Reply:
Thanks for thinking of this! `wavefront.NewClient` checks for the `nil` case and uses a default `http.Client` in that case - I think the result is reasonable; it'll error out if you try to connect to an `https` server that would require manual TLS configuration, and works with an `http` server or an `https` server that uses a certificate that the OS would already trust.

Maybe in that last case it should error out, since it seems odd to say `https` and also `tls_enable=false`?


Here's an example of the error message when `tls_enable=false` and the target server has a cert signed by an internal CA

```
2023-06-02T15:21:25Z I! [agent] Hang on, flushing any cached metrics before shutdown
2023-06-02T15:21:25Z I! error reporting to Wavefront. buffering lines.
2023-06-02T15:21:25Z E! [agent] Error writing to outputs.wavefront: error reporting wavefront format data to Wavefront: "Post \"https://localhost:9999/report?f=wavefront\": tls: failed to verify certificate: x509: “localhost” certificate is not trusted"
2023-06-02T15:21:25Z I! [agent] Stopping running outputs
2023-06-02T15:21:25Z I! error reporting to Wavefront. buffering lines.
2023-06-02T15:21:25Z I! error reporting wavefront format data to Wavefront: "Post \"https://localhost:9999/report?f=wavefront\": tls: failed to verify certificate: x509: “localhost” certificate is not trusted"
```